spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1574,24 +1574,37 @@ def append(self, text):
     @Slot()
     def paste(self):
         """
+        Insert text or file/folder path copied from clipboard.
+
         Reimplement QPlainTextEdit's method to fix the following issue:
         on Windows, pasted text has only 'LF' EOL chars even if the original
-        text has 'CRLF' EOL chars
+        text has 'CRLF' EOL chars.
+        This function changes the clipboard data if they are copied as
+        files/folder but does not change normal text data except if they are
+        multiple lines. Since we are changing clipboard data we can not use
+        paste, which directly pastes from clipboard instead we use
+        insertPlainText and pass the formatted/changed text.
         """
         clipboard = QApplication.clipboard()
-        # This is here to prevent pasting mime data in the Editor.
-        # See issue 8566 for the details.
+        text = to_text_string(clipboard.text())
         if clipboard.mimeData().hasUrls():
-            return
-        else:
-            text = to_text_string(clipboard.text())
-            if len(text.splitlines()) > 1:
-                eol_chars = self.get_line_separator()
-                text = eol_chars.join((text + eol_chars).splitlines())
-                clipboard.setText(text)
-            # Standard paste
-            TextEditBaseWidget.paste(self)
-            self.document_did_change(text)
+            # Have copied file and folder urls pasted as text paths.
+            # See PR: #8644 for details.
+            urls = clipboard.mimeData().urls()
+            if all([url.isLocalFile() for url in urls]):
+                if len(urls) > 1:
+                    sep_chars = ',' + self.get_line_separator()
+                    text = sep_chars.join('"' + url.toLocalFile().
+                                          replace(osp.os.sep, '/')

```

Comment:
@Khalilsqu I don't remember why we replaced the separators here and not in the case of the single url?

Reply:
Thanks, you are right. It was suggested by @CAM-Gerlach  in another PR to standarize the seperator in all OS into '/', I just forgot to make the same change for a single file path. I will take care of it soon.